pkg/views/target/list/view.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package target
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/daytonaio/daytona/pkg/serverapiclient"
+
+	"github.com/charmbracelet/bubbles/table"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"golang.org/x/term"
+)
+
+var columns = []table.Column{
+	{Title: "Provider Name", Width: 30},
+	{Title: "Container Image", Width: 30},
+	{Title: "Sock Path", Width: 30},
+}
+
+type Option struct {
+	ContainerImage string `json:"Container Image"`
+	SockPath       string `json:"Sock Path"`
+	// other fields
+}
+
+type model struct {
+	table       table.Model
+	initialRows []table.Row
+}
+
+func (m model) Init() tea.Cmd {
+	return tea.Quit
+}
+
+func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var cmd tea.Cmd
+	switch msg := msg.(type) {
+	case tea.WindowSizeMsg:
+		rows, cols := getRowsAndCols(msg.Width, m.initialRows)
+		m.table = getTable(rows, cols, m.table.Cursor())
+		return m, nil
+	}
+	m.table, cmd = m.table.Update(msg)
+	return m, cmd
+}
+
+var baseStyle = lipgloss.NewStyle().
+	BorderStyle(lipgloss.RoundedBorder()).
+	Padding(0, 1).MarginBottom(1)
+
+func (m model) View() string {
+	return baseStyle.Render(m.table.View())
+}
+
+func getTable(rows []table.Row, cols []table.Column, activeRow int) table.Model {
+	t := table.New(
+		table.WithColumns(cols),
+		table.WithRows(rows),
+		table.WithHeight(len(rows)),
+	)
+
+	style := table.DefaultStyles()
+	style.Header = style.Header.
+		BorderStyle(lipgloss.NormalBorder()).
+		BorderBottom(true).
+		AlignHorizontal(lipgloss.Left)
+
+	style.Selected = style.Selected.
+		Foreground(style.Cell.GetForeground()).
+		Background(style.Cell.GetBackground()).
+		Bold(false)
+
+	t.SetStyles(style)
+	t.SetCursor(activeRow)
+
+	return t
+}
+
+func getRowsAndCols(width int, initialRows []table.Row) ([]table.Row, []table.Column) {
+	colWidth := 0
+	cols := []table.Column{}
+
+	for _, col := range columns {
+		if colWidth+col.Width > width {
+			break
+		}
+
+		colWidth += col.Width
+		cols = append(cols, col)
+	}
+
+	rows := []table.Row{}
+	for _, row := range initialRows {
+		rows = append(rows, row[:len(cols)])
+	}
+
+	return rows, cols
+}
+
+func ListTargets(targets []serverapiclient.ProviderTarget) {
+	m := renderTargetList(targets)
+
+	_, err := tea.NewProgram(m).Run()
+	if err != nil {
+		fmt.Println("Error running program:", err)
+		os.Exit(1)
+	}
+}
+
+func renderTargetList(targets []serverapiclient.ProviderTarget) model {
+	rows := []table.Row{}
+
+	for _, target := range targets {
+		var options Option
```

Comment:
@harkiratsm thanks for the PR!

Good effort on trying to format the table in this way, but unfortunately, this is not what we want.

Provider targets should not be typed in Daytona. Provider targets are typed only in the provider and are passed between them as JSON strings. This means that Daytona can not have a guarantee of the type of the targets options JSON object. Additionally, when a new provider is added with new target options, this would fail completely.

I suggest that the table has the following columns `Provider (provider name) | Target (target name) | Options (target options JSON string)`.

This way, there is no assumption on the type of the target and this will work for any provider in the future.

One last thing, you should group targets based on provider name so that, for example, Docker provider targets are always together, AWS provider targets are always together, etc.

Reply:
Understood, here's the revised table.

![image](https://github.com/daytonaio/daytona/assets/71957674/8ba00d32-93a3-4809-ad03-7193636f46b8)

Reply:
Looks good to me!